sdk/core/src/errors.rs

```diff
@@ -42,6 +42,8 @@ pub enum Error {
     Json(#[source] serde_json::Error),
     #[error("Other error")]
     Other(#[source] Box<dyn std::error::Error + Send + Sync + 'static>),
+    #[error("Authorization policy error")]
+    AuthorizationPolicy(String),
```

Comment:
```suggestion
    #[error("authorization policy error: {0}")]
    AuthorizationPolicy(Box<dyn std::error::Error + Send + Sync + 'static>),
```

Nit: can you also move this above the `Other` variant so that `Other` is last?

Reply:
Switched the variants, but was not able to work with the boxed error. See comment below.